### DISCUSSION TITLE
How to include dependencies in Cake Addin? Dll not found when running task

### DISCUSSION THREAD

**Discussion by Toyonut** [cake-build/cake#3885] [Extension Q&A] [answered]
I have a Cake addin I have been working on to try and set AssemblyInfo.cs and csproj properties https://github.com/Toyonut/Cake.AssemblyInfoSetter. I am dealing with an issue of getting file locks when trying to use a regex replace to set assemblyinfo.cs properties running Cake under Jenkins. I thought it would be good to try and write an addin and have something like Polly retry the file save operation.  I did a dotnet pack in the build.cake and uploaded the resulting nupkg to our internal package repo. It gets restored by the Cake script but then fails with an error similar to unable to find Polly.dll. 
I have been looking through repos like https://github.com/cake-contrib/Cake.AWS.ElasticBeanstalk but they don't seem to be doing anything different when referencing libraries to include the DLLs. 

What is the recommended way to include referenced DLLs in the nupkg file? Is there a different way I should be approaching this?

**Comment by pascalberger** [comment, marked as answer]
There are two possible options:

1. If license allows you can deploy dependent assemblies as part of your addin NuGet package. This will create a self-contained addin package which will "just work" with either Cake Scripting or Cake Frosting (as long as there are no other addins requiring or bringing in a different version of the same assemblies)
2. You can define dependencies on the NuGet package. When used with Cake Scripting the addin would need to be loaded using the [loaddependencies=true parameter](https://cakebuild.net/docs/writing-builds/preprocessor-directives/addin#dependencies). With Cake Frosting it will just work, including compile time warnings if different addins require incompatible dependencies.

**Reply by Toyonut** [reply]
Thanks very much @pascalberger I appreciate the help. Will use the second option next time.
